.testhelpers.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+create_type() {
+  local type=$1
+  local path=$2
+  local type_name=$3
+  local image=$4
+  local args=$5
+  local runtime=${6:-core}
+
+  echo "Create $type $type_name"
+
+  pushd $path
+    if [ -e '.fats/create' ]; then
+      args="${args} `cat .fats/create`"
+    fi
+
+    # create function/application
+    fats_echo "Creating $type_name:"
+    riff $type create $type_name $args --image $image --namespace $NAMESPACE --tail &
+    riff $runtime deployer create $type_name --$type-ref $type_name --namespace $NAMESPACE --tail
+
+    # TODO reduce/eliminate this sleep
+    sleep 5
+  popd
+}
+
+invoke_type() {
```

Comment:
`invoke` is strongly associated with functions. It seems like you're hacking around the invoke behavior for applications. What if we keep this method focused on resolving the ingress/hostname and have it receive a single `curl_opts` param?

For functions the curl_opts param would be `"-H 'Content-Type: text/plain' -d $input_data"`

We can decide what to do for applications. Although, using `application/x-www-form-urlencoded` as the spec for applications seems like an odd choice.

Reply:
done.